data_highlight/highlighter.py

```diff
@@ -1,6 +1,28 @@
 from .support.char_array import CharIndex
 from .support.line import Line
 from .highlighter_functionality.export import export as export_from_functionality
+from .support.token import Token
+
+
+def combine(*args):
+    for i in args:
+        if not isinstance(i, Token):
+            print("Not Token  argument was passed")
+            exit(1)
+
+    first_token = args[0]
+    arr_small=[]
+    for index in range(1,len(args)):
+        arr_Small_tokens=args[index].array_of_tokens
+        for index_small_token in arr_Small_tokens:
+            arr_small.append(index_small_token)
+
+
+    for i in arr_small:
+        first_token.array_of_tokens.append(i)
+
+
+    return first_token
```

Comment:
Thanks for that @Romucmad - I've been quite relaxed about type checking.  Maybe this is a case where it's justified.  Other than that, my approach was similar to yours, though I found that `extend()` adds the individual items to the other array, rather than adding the whole list:

![image](https://user-images.githubusercontent.com/1108513/68654839-b910bf00-0526-11ea-99cf-c35363ac33d4.png)

Oh, I also avoided modifying the first token - so it can still be used to highlight where just it has been used.  If we have `Date` and `Time` tokens, we may combine them for some one parser for this file, and we may wish to leave them separate for another parser on this file.